src/core/trtserver2.h

```diff
@@ -297,6 +298,25 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER2_InferenceRequestOutputData(
     const void** base, size_t* byte_size, TRTSERVER_Memory_Type* memory_type,
     int64_t* memory_type_id);
 
+/// Get the indices of all unique models in the model repository.
```

Comment:
Follow pattern from trtserver.h, each API object has section, so above have:
//
// TRTSERVER2_ModelIndex
//
with TRTSERVER2_ModelIndexDelete(). You also need a function to get the actual index strings out of the object. Like TRTSERVER2_ModelIndexNames to return pointers to all the names. The returned pointers will be owned by the object and so the caller cannot modify or delete. See description for TRTSERVER_ServerExtensions.
You do not need a New because the only way to create is by the server call below.